### PR TITLE
fix(devices): handle Shelly NG wizard polling edge cases

### PR DESCRIPTION
**Pull request by akadlec**
## Summary

- Preserve manually verified Shelly NG passwords within a discovery session so later mDNS rediscovery does not downgrade protected ready devices back to password-required.
- Promote admin polling placeholders when a discovered device transitions from `checking` to `ready`, restoring default selection, suggested category, and friendly name.
- Add regression tests for both review-reported wizard edge cases.

## Validation

- `pnpm --filter @fastybird/smart-panel-backend exec jest src/plugins/devices-shelly-ng/services/shelly-ng-discovery.service.spec.ts`
- `pnpm --filter @fastybird/smart-panel-admin exec vitest run src/plugins/devices-shelly-ng/composables/useDevicesWizard.spec.ts`
- `pnpm --filter @fastybird/smart-panel-backend run type-check`
- `pnpm --filter @fastybird/smart-panel-admin run type-check`
- `pnpm --filter @fastybird/smart-panel-backend exec eslint src/plugins/devices-shelly-ng/services/shelly-ng-discovery.service.ts src/plugins/devices-shelly-ng/services/shelly-ng-discovery.service.spec.ts`
- `pnpm --filter @fastybird/smart-panel-admin exec eslint src/plugins/devices-shelly-ng/composables/useDevicesWizard.ts src/plugins/devices-shelly-ng/composables/useDevicesWizard.spec.ts`
- `git diff --check`

Follow-up to #599.

---

> [!NOTE]
> **Medium Risk**
> Touches discovery state management and credential reuse within an in-memory session; mistakes could cause incorrect device selection defaults or failure to re-authenticate rediscovered devices.
> 
> **Overview**
> Improves Shelly NG device discovery/wizard resiliency around rediscovery and polling transitions.
> 
> On the **backend**, discovery sessions now persist manually verified passwords (keyed by hostname and device identifier) and reuse them for later mDNS rediscovery so protected devices don’t regress back to `needs_password`; `inspectDevice` now returns the discovered snapshot to enable password caching only when a device is actually `ready`.
> 
> On the **admin wizard**, session application logic now "promotes" placeholder devices that move from `checking` to `ready` by restoring default selection, suggested category, and friendly name *without* overriding explicit user deselection after a device has already been ready once. New unit tests cover these edge cases on both frontend and backend.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 27cf824482cd819cacf1db4198536da8776f96cc. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: apps/admin/src/plugins/devices-shelly-ng/composables/useDevicesWizard.ts [hunk 1]
```diff
@@ -99,18 +99,26 @@ export const useDevicesWizard = (): IUseDevicesWizard => {
...
+
 		session.value = nextSession;
 
 		for (const device of nextSession.devices) {
-			if (selected[device.hostname] === undefined) {
+			const previousDevice = previousDevices.find((item) => item.hostname === device.hostname);
+			const becameReady = previousDevice?.status === 'checking' && device.status === 'ready';
+
+			if (selected[device.hostname] === undefined || becameReady) {
 				selected[device.hostname] = device.status === 'ready';
```
[the remark this answers is not in the document]
Comment (akadlec) [reply apps/admin/src/plugins/devices-shelly-ng/composables/useDevicesWizard.ts]: Fixed in e12b0df8d: `applySession` now tracks hostnames that have already reached `ready` and only auto-selects the first `checking` -> `ready` promotion. If the user deselects an already-ready device and rediscovery briefly moves it through `checking`, the later `ready` snapshot preserves that deselection. Added an admin regression test for this flow.

File: apps/backend/src/plugins/devices-shelly-ng/services/shelly-ng-discovery.service.ts [hunk 2]
```diff
@@ -98,10 +99,13 @@ export class ShellyNgDiscoveryService {
...
 			discoverer,
 			devices: new Map(),
+			passwords: new Map(),
 		};
 
 		discoverer.on('discover', (device: { deviceId?: string; hostname?: string }) => {
-			void this.inspectDevice(session, device.hostname ?? device.deviceId ?? '', 'mdns', null);
+			const hostname = device.hostname ?? device.deviceId ?? '';
+
+			void this.inspectDevice(session, hostname, 'mdns', session.passwords.get(hostname) ?? null);
```
[the remark this answers is not in the document]
Comment (akadlec) [reply apps/backend/src/plugins/devices-shelly-ng/services/shelly-ng-discovery.service.ts]: Fixed in e12b0df8d: verified manual credentials are now stored under both the manual hostname and the discovered Shelly identifier. When an mDNS event falls back to `deviceId` because `hostname` is missing, the rediscovery still finds and reuses the verified password. Updated the backend regression test to cover a rediscovery event without `hostname`.

File: apps/admin/src/plugins/devices-shelly-ng/composables/useDevicesWizard.ts [hunk 3]
```diff
@@ -99,18 +99,26 @@ export const useDevicesWizard = (): IUseDevicesWizard => {
...
+		const previousDevices = session.value?.devices ?? [];
+
 		session.value = nextSession;
 
 		for (const device of nextSession.devices) {
-			if (selected[device.hostname] === undefined) {
+			const previousDevice = previousDevices.find((item) => item.hostname === device.hostname);
+			const becameReady = previousDevice?.status === 'checking' && device.status === 'ready';
+
+			if (selected[device.hostname] === undefined || becameReady) {
```
[the remark this answers is not in the document]
Comment (akadlec) [reply apps/admin/src/plugins/devices-shelly-ng/composables/useDevicesWizard.ts]: Fixed in e12b0df8d: the wizard now preserves user deselection for devices that were already ready before a rediscovery cycle. Auto-selection is limited to the first time a placeholder transitions from `checking` to `ready`, and the new admin test covers ready -> checking -> ready after deselection.

File: apps/backend/src/plugins/devices-shelly-ng/services/shelly-ng-discovery.service.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (akadlec) [reply apps/backend/src/plugins/devices-shelly-ng/services/shelly-ng-discovery.service.ts]: Fixed in 27cf82448: the mDNS callback now resolves the password by the emitted hostname first, then falls back to `deviceId` before inspecting the device. The regression test now covers manual verification by IP followed by rediscovery with a different DNS hostname plus the same Shelly identifier.
